index/scorch/segment/zap/contentcoder.go

```diff
@@ -111,6 +117,15 @@ func (c *chunkedContentCoder) flushContents() error {
 	c.final = append(c.final, c.compressed...)
 
 	c.chunkLens[c.currChunk] = uint64(len(c.compressed) + len(metaData))
+
+	if c.writeOutInChunks {
+		_, err := c.w.Write(c.final)
+		if err != nil {
+			return err
+		}
+		c.final = nil
```

Comment:
this nil'ing will cause memory allocations the next time somebody appends() to final.

instead, should zero out the slice length via c.final = c.final[:0]

Reply:
Done.